### DISCUSSION TITLE
What's the underlying algorithm of `min_weight_matching`?

### DISCUSSION THREAD

**Discussion by eRuaro** [networkx/networkx#5969] [Q&A]
Hello! I've been reading up on minimum weights matching and I saw that NetworkX has a function for it. What's the underlying algorithm? And how does it differ from say the `Hungarian Algorithm`?

**Comment by rossbar** [comment]
The matching functions can be found in the `matching.py` module:

https://github.com/networkx/networkx/blob/5c0b11afb4c0882a070d522ef3fa41482ba935d3/networkx/algorithms/matching.py

A quick look there shows that the `min_weight_matching` function actually uses the `max_weight_matching` under-the-hood (after some munging of the weights). The `max_weight_matching` implementation has the following:

https://github.com/networkx/networkx/blob/5c0b11afb4c0882a070d522ef3fa41482ba935d3/networkx/algorithms/matching.py#L387-L404
